File: 백준/구현/1316 그룹단어 체커.py
# 방법 1
wordsNum = int(input())
groupWordCnt = 0

for i in range(wordsNum):
  word = input()
  error = 0 
  for index in range(len(word)-1):
    if word[index] != word[index+1]: # 문자열이 연달아있는 경우 신경쓸 필요가 없다 
      new_word = word[index+1:]
      if new_word.count(word[index]) > 0:
        error += 1

  if error == 0:
    groupWordCnt += 1
print(groupWordCnt)

# 방법 2
wordsNum = int(input())
groupWordCnt = wordsNum

for i in range(wordsNum):
  word = input()
  for j in range(len(word)-1):
    if word[j] == word[j+1]:
      pass
    elif word[j] in word[j+1:]:
      groupWordCnt -= 1
      break
print(groupWordCnt)

File: 백준/구현/1316 그룹단어 체커.js
// 방법 1

let fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const count = Number(input[0]);

let answer = 0;

for (i = 1; i <= count; i++) {
  let word = input[i];
  let flag = true;
  let used = [];
  for (let j = 0; j < word.length; j++) {
    const char = word[j];
    if (!used.includes(char)) {
      used.push(char);
    } else if (char !== word[j - 1]) {
      flag = false;
      break;
    }
  }
  if (flag) answer++;
}

console.log(answer);

// 방법 2

const fs = require("fs");
const input2 = fs.readFileSync("/dev/stdin").toString().split("\n");

const N = Number(input2[0]);
let count2 = N;
let word;

for (let i = 1; i <= N; i++) {
  word = input2[i];
  for (let j = 0; j < word.length - 1; j++) {
    if (word[j] != word[j + 1] && !!word.slice(j + 1).includes(word[j])) {
      count2--;
      break;
    }
  }
}
console.log(count2);
